### PR TITLE
feat: add contentful reqs and util

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
[PROD-3023](https://2u-internal.atlassian.net/browse/PROD-3023)
Adds Contentful requirements and utility function to read data from there.

### Testing Instructions
1. Run python shell in discovery shell: `make discovery-shell` `./manage.py shell`
2. Set the following variables in `private.py` file:
```
CONTENTFUL_SPACE_ID = <space_id>
CONTENTFUL_CONTENT_DELIVERY_API_KEY = <api_key>
```
3. Run the following code in shell to read the data from contentful:
```
from course_discovery.apps.course_metadata.utils import get_data_from_contentful
get_data_from_contentful('degreeDetailPage')
```
Expect to see a response like `<Entry[degreeDetailPage] id='3BjaNC0mKiqCtsuaLKCSPC'>`

### CODE REVIEW

File: course_discovery/apps/course_metadata/utils.py [hunk 1]
```diff
@@ -847,3 +848,19 @@ def transform_skills_data(skills_data):
...
+    """
+    Utility function to get data from contentful. Returns contentful entries of the content_type.
+
+    Args:
+        content_type (str): Contentful table-like instance comprised of fields.
+    """
+    client = Client(
+        settings.CONTENTFUL_SPACE_ID,
+        settings.CONTENTFUL_CONTENT_DELIVERY_API_KEY
+    )
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/utils.py:863]: Does this establish some connection with Contentful or is it just for instantiation?

**Review by DawoudSheraz — APPROVED** [review]
Looks good. I have not tested it on local.

File: course_discovery/apps/course_metadata/utils.py (hunk 1 above)
Comment (Ali-D-Akbar) [reply course_discovery/apps/course_metadata/utils.py:863]: It makes a connection just like any other client-like object in Python.

**Review by AliAdnanSohail — APPROVED** [review]

File: course_discovery/apps/course_metadata/utils.py [hunk 2]
```diff
@@ -847,3 +848,19 @@ def transform_skills_data(skills_data):
...
+    return client.entries({
```
Comment (ansabgillani) [inline course_discovery/apps/course_metadata/utils.py:864]: just use return 
`return Client( 
        settings.CONTENTFUL_SPACE_ID,
        settings.CONTENTFUL_CONTENT_DELIVERY_API_KEY
    ).entries({
            'content_type': content_type
    })
 `

**Review by ansabgillani — APPROVED** [review]

**Pull request merged**